### PR TITLE
Add formId to actions

### PR DESCRIPTION
**Pull request by dododedodonl**
## Description

Continuation of #12091 

## Visual changes

None

## Functional changes

- [X] Code style has been fixed by running the `composer cs` command.
  - `npm run prettier` suggests quite some changes to untouched files, I did not commit those
- [x] Changes have been tested to not break existing functionality.
- [x] Documentation is up-to-date.

### CODE REVIEW

**Comment by dododedodonl** [comment]
https://github.com/filamentphp/filament/pull/12091#issuecomment-2027815214
> (...)
> theres no real need to introduce an ID or wire:key on the form.
> (...)

@danharrin The `wire:key` was already there. The `form=""` attribute has to contain the ID of the form to submit, so for this to work in a panel, the form has to have an ID. Is adding a default one, equal to the `wire:key`, a valid option?

File: packages/actions/src/Concerns/CanSubmitForm.php [hunk 1]
```diff
@@ -25,4 +27,16 @@ public function getFormToSubmit(): ?string
...
     }
+
+    public function formId(?string $id): static
+    {
+        $this->formId = $id;
+
+        return $this;
+    }
+
+    public function getFormid(): ?string
```
Comment (danharrin) [inline packages/actions/src/Concerns/CanSubmitForm.php]:
```suggestion
    public function getFormId(): ?string
```

**Comment by danharrin** [comment]
> The wire:key was already there

My mistake

> for this to work in a panel, the form has to have an ID

For user friendliness, we can probably just assign the edit/create forms an ID of `form`

**Comment by danharrin** [comment]
Thanks

**Pull request merged**